### DISCUSSION TITLE
How to use custom aligning relative to margins

### DISCUSSION THREAD

**Discussion by th0mr** [brechtm/rinohtype#402] [Q&A] [answered]
I am currently trying to achieve this styling for my tables, where the table is centered to the content but still allows the table to spill over left by 2cm past the content when the table needs the extra space. i.e centering around where it usually centers + 2cm (this may be clearer when looking at the two examples below)

Below is a mock up of how i want it to look.
![desired_behavior](https://user-images.githubusercontent.com/87852659/222115317-d91bb2da-61f3-49bc-8552-98ac8a9748c3.png)

As you can see, the larger table requires the use of the whole area the table has access to, i.e. the content area + 2cm to the left
Whereas, a smaller table that does not require this extra 2cm is centered within the content area in line with how the paragraph is centered.

Below is what it currently looks like with my stylesheet. 
![current_behavior](https://user-images.githubusercontent.com/87852659/222115347-0df58e12-c6db-4d22-b663-f02d9357427a.png)

Here is the current style sheet I'm using.

```
[STYLESHEET]
name=min_example
description=Minimal example of table problem
pygments_style=friendly
base=sphinx

[chapter]
# All content needs to be pushed 2cm right
margin_left=2cm

[heading level 1]
# Set margin to 0cm
margin_left=0cm
# adds tab character between tab stop and heading to create the gap
before="\t"
# set tab stop at the margin
tab_stops=0cm LEFT
# create room for the section label
indent_first=-2cm

[table]
margin_left=-2cm
horizontal_align=center
split_minimum_rows=2
```

I've tried a few things in the stylesheet and have messed around with overriding the _align() method in Flowable, but I am not having much success.

Is there anything that could be done to to achieve this? I appreciate that custom centering like this may be a niche request, so I would understand if this is beyond the capabilities of rinohtype at the moment.

Any advice would be greatly appreciated 😄

**Comment by brechtm** [comment, marked as answer]
I can see merit in supporting this, but I'm not convinced rinohtype should provide extra style properties for such a specific case...

You can however achieve this with some more manual intervention by defining a _wide table_ style definition for which you set a left margin of -2cm. This _wide table_ style should match tables for which e.g. the _wide_ class is set.

I haven't tested this, but it should be a good start:
```ini
[wide table : Table(has_class='wide')]
base = table
margin_left = -2cm
# optionally: width = 100%
```

```rst
.. table:: Wide table
   :class: wide

   +------------------------+------------+----------+----------+
   | Header row, column 1   | Header 2   | Header 3 | Header 4 |
   | (header rows optional) |            |          |          |
   +========================+============+==========+==========+
   | body row 1, column 1   | column 2   | column 3 | column 4 |
   +------------------------+------------+----------+----------+
   | body row 2             | Cells may span columns.          |
   +------------------------+------------+---------------------+
   | body row 3             | Cells may  | - Table cells       |
   +------------------------+ span rows. | - contain           |
   | body row 4             |            | - body elements.    |
   +------------------------+------------+----------+----------+
   | body row 5             | Cells may also be     |          |
   |                        | empty: ``-->``        |          |
   +------------------------+-----------------------+----------+
```

Of course, this isn't yet documented properly. In the meantime, you can have a look at the _table*_ examples in https://github.com/brechtm/rinohtype/tree/master/tests_regression/rst.

**Reply by th0mr** [reply]
I agree it is too specific of a case to be supported by rinohtype.

Thank you for your advice, I'll look into the wide tables now.
